code/modules/reagents/chemistry/machinery/reagentgrinder.dm

```diff
@@ -520,11 +521,20 @@
 		if(beaker.reagents.holder_full())
 			return
 
-	// Everything else - Transfers reagents from it into beaker
-	for(var/obj/item/reagent_containers/O in holdingitems)
-		O.reagents.trans_to(beaker, O.reagents.total_volume)
+	// Cigarettes
+	for(var/obj/item/clothing/mask/cigarette/O in holdingitems)
+		if(reagent_container_grind(O))
+			return
 
-		if(!O.reagents.total_volume)
-			remove_object(O)
-		if(beaker.reagents.holder_full())
+	// Pills and Patches
```

Comment:
This is a bunch more than just pills and patches, please keep the comment as it used to be